query/src/main/scala/filodb/query/exec/StitchRvsExec.scala

```diff
@@ -12,41 +12,55 @@ import filodb.query.Query.qLogger
 
 object StitchRvsExec {
 
-  def stitch(v1: RangeVector, v2: RangeVector): RangeVector = {
-    val rows = StitchRvsExec.merge(Seq(v1.rows(), v2.rows()))
-    IteratorBackedRangeVector(v1.key, rows, RvRange.union(v1.outputRange, v2.outputRange))
-  }
+  private class RVCursorImpl(vectors: Iterable[RangeVectorCursor], outputRange: Option[RvRange])
+    extends RangeVectorCursor {
+    private val bVectors = vectors.map(_.buffered)
+    val mins = new mutable.ArrayBuffer[BufferedIterator[RowReader]](2)
+    val nanResult = new TransientRow(0, Double.NaN)
+    val tsIter: BufferedIterator[Long] = outputRange match {
+      case Some(RvRange(startMs, stepMs, endMs)) =>
+        Iterator.iterate(startMs){_ + stepMs}.takeWhile( _ <= endMs).buffered
+      case None                                  => Iterator.iterate(Long.MaxValue)(_ =>Long.MaxValue).buffered
+    }
+    override def hasNext: Boolean = outputRange match {
+                // In case outputRange is provided, the merging will honor the provided range, that is, the
+                // result of metrging will have a range provided by outputRange. In case outputRange is not provided
+                // i.e. None, then hasNext should be determined by the vectors it merges. The none handling is just done
+                // for matching both possibilities and in reality for a given periodic series with a range, the
+                // outputRange should never be none
+                case Some(_)      => tsIter.hasNext
+                case None         => bVectors.exists(_.hasNext)
+    }
 
-  def merge(vectors: Iterable[RangeVectorCursor]): RangeVectorCursor = {
-    // This is an n-way merge without using a heap.
-    // Heap is not used since n is expected to be very small (almost always just 1 or 2)
-    new RangeVectorCursor {
-      val bVectors = vectors.map(_.buffered)
-      val mins = new mutable.ArrayBuffer[BufferedIterator[RowReader]](2)
-      val nanResult = new TransientRow(0, Double.NaN)
-      override def hasNext: Boolean = bVectors.exists(_.hasNext)
-
-      override def next(): RowReader = {
-        mins.clear()
-        var minTime = Long.MaxValue
-        bVectors.foreach { r =>
-          if (r.hasNext) {
-            val t = r.head.getLong(0)
-            if (mins.isEmpty) {
-              minTime = t
-              mins += r
-            }
-            else if (t < minTime) {
-              mins.clear()
-              mins += r
-              minTime = t
-            } else if (t == minTime) {
-              mins += r
-            }
+    override def next(): RowReader = {
+
```

Comment:
Retain old comment:
```
// This is an n-way merge without using a heap.
// Heap is not used since n is expected to be very small (almost always just 1 or 2)
```